core/src/test/scala/org/apache/spark/executor/ExecutorSuite.scala

```diff
@@ -270,6 +272,28 @@ class ExecutorSuite extends SparkFunSuite
     heartbeatZeroAccumulatorUpdateTest(false)
   }
 
+  test("SPARK-39696: Using accumulators should not cause heartbeat to fail") {
+    val conf = new SparkConf().setMaster("local").setAppName("executor suite test")
+    conf.set(EXECUTOR_HEARTBEAT_INTERVAL.key, "1ms")
+    sc = new SparkContext(conf)
+
+    val accums = (1 to 10).map(i => sc.longAccumulator(s"mapperRunAccumulator${i}"))
+    val input = sc.parallelize(1 to 10, 10)
+    val baseRdd = input.map(i => (i, i))
+    var testRdd = baseRdd
+    (0 to 10).foreach( i =>
+        testRdd = testRdd.map(x => {accums.map(_.add(1)); (x._1 * i, x._2)}).reduceByKey(_ + _)
```

Comment:
1. can be`accums.foreach(_.add(1))`
2. should add spaces after `{` and before `}`, like

```
{ accums.map(_.add(1)); (x._1 * i, x._2) }
```